### DISCUSSION TITLE
??= Operator

### DISCUSSION THREAD

**Discussion by chuck-flowers** [dotnet/csharplang#725] [General]
I apologize if this has already been addressed but I could not find any issue dealing with it. I have found myself wanting this operator more and more recently. 

A few years ago in C# we had to write this
```
if(foo == null)
   foo = new Foo();
```

Now we can write
```
foo = foo ?? new Foo();
```

But it can still get very lengthy with longer descriptive variable and type names
```
aDescriptiveFooVarName = aDescriptiveFooVarName ?? new DescriptiveFooClass();
```

What I would like is a compound assignment operator that can be used like so
```
aDescriptiveFooVarName ??= new DescriptiveFooClass();
```

I feel like it more concisely conveys intent and that it is a common enough pattern it's deserving of a language feature. If this is considered an antipattern of some sort with a better solution, please feel free to let me know because I'm getting tired of typing the same variable names back to back like this.

**Comment by cpplearner** [comment]
duplicate of #34 ?

**Comment by jnm2** [comment]
Yes, this is a dupe.

**Comment by jnm2** [comment]
Great minds think alike, and sometimes so do ours.

(However, great minds use search engines. 😉)

**Comment by chuck-flowers** [comment]
I must not have been searching with the right words. Either way I'll check out #34.

**Comment by jnm2** [comment]
@chuck-flowers Oh, it happens to me too. Just did the same thing myself. 😳
